### PR TITLE
Tests with Aqua.jl

### PR DESCRIPTION
**Pull request by lbenet**

### CODE REVIEW

**Comment by lbenet** [comment]
Tests are now passing in nightly too

**Comment by PerezHz** [comment]
Many thanks for adding this @lbenet! Is this PR still WIP, or is it ready for review? (actually changes look pretty much straight-forward so far)

**Comment by lbenet** [comment]
It is essentially ready; I don't think there are too many additions to make. Also, I don't know if we should tag a new patch version...

File: test/aqua.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+@testset "Aqua tests (additional)" begin
+    Aqua.test_undefined_exports(TaylorIntegration)
+    Aqua.test_deps_compat(TaylorIntegration)
+    Aqua.test_stale_deps(TaylorIntegration;
+        ignore=[:DiffEqBase, :RecursiveArrayTools, :Requires, :StaticArrays])
+    Aqua.test_piracies(TaylorIntegration)
+    Aqua.test_unbound_args(TaylorIntegration)
+    Aqua.test_project_extras(TaylorIntegration)
+    Aqua.test_persistent_tasks(TaylorIntegration)
+end
```
Comment (PerezHz) [inline test/aqua.jl:35]: Took a quick look to the Aqua docs, seems like there's a do-it-all function here, but I this is exactly equivalent to call the tests one by one as is it done now?
```suggestion
@testset "Aqua tests (additional)" begin
    Aqua.test_all(
    TaylorIntegration;
    stale_deps=(ignore=[:DiffEqBase, :RecursiveArrayTools, :Requires, :StaticArrays],),
    )
end
```

**Review by PerezHz — APPROVED** [review]
Just left a minor suggestion which I think is equivalent (though perhaps a bit more compact) to what you're already proposing. Either way, LGTM!

**Comment by PerezHz** [comment]
> Also, I don't know if we should tag a new patch version...

I would actually agree to bump the patch version

File: test/aqua.jl (hunk 1 above)
Comment (lbenet) [reply test/aqua.jl]: Committed your suggestion; thanks!

**Comment by lbenet** [comment]
Merging...

**Pull request merged**